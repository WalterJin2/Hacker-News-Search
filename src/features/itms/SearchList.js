import React from "react";
import { FaUserAlt } from "react-icons/fa";
import { formatDistanceToNow } from "date-fns";
export default function SearchList({ itms }) {
  return (
    <div className="flex flex-col justify-between items-center  w-full px-1 py-1 border border-green-800">
      {itms.hits.map((data) => {
        const {
          title,
          author,
          points,
          created_at,
          num_comments,
          url,
          objectID,
        } = data;

        return (
          <div className="flex flex-col w-full px-2  py-4 gap-2" key={objectID}>
            <a className="resultTag  px-1 py-1" href={url}>
              <h1
                className="px-1 py-1"
                dangerouslySetInnerHTML={{
                  // Assume the response is sanitised
                  __html:
                    (data._highlightResult.title &&
                      data._highlightResult.title.value) ||
                    (data._highlightResult.story_title &&
                      data._highlightResult.story_title.value) ||
                    "Title not found",
                }}
              ></h1>
            </a>
            <div className="flex justify-start gap-1  pt-1 pb-4 border-b border-green-900">
              <div className="resultTag px-1 py-1">{points} points</div>
              <div className="flex justify-between items-center resultTag px-1 py-1 ">
                <div className="px-1 ">
                  <FaUserAlt />
                </div>
                <div className="px-1 ">{author}</div>
              </div>
              <div className="resultTag  px-1 py-1">
                {formatDistanceToNow(new Date(created_at), {
                  addSuffix: true,
                })}
              </div>
              <div className="resultTag px-1 py-1">{num_comments} comments</div>
              <a className="resultTag  px-1 py-1" href={url}>
                {url}
              </a>
            </div>
            {data._highlightResult.comment_text ||
            data._highlightResult.story_text ? (
              <div
                dangerouslySetInnerHTML={{
                  __html:
                    (data._highlightResult.comment_text &&
                      data._highlightResult.comment_text.value) ||
                    (data._highlightResult.story_text &&
                      data._highlightResult.story_text.value),
                }}
              />
            ) : null}
          </div>
        );
      })}
    </div>
  );
}
