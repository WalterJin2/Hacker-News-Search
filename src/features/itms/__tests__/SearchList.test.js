const searchResult = require('../../../test/searchTestsMockData.json').searchResult
import SearchList from '../SearchList';
import { renderWithProviders } from "../../../test/test-utils";

it('SeachList renders correctly', () => {
    const props = {
      itms: searchResult,
    };
  const { asFragment } = renderWithProviders(<SearchList {...props} />);
  expect(asFragment()).toMatchSnapshot();
  
});