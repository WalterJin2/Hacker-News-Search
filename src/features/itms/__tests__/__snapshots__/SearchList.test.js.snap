// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeachList renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-between items-center  w-full px-1 py-1 border border-green-800"
  >
    <div
      class="flex flex-col w-full px-1 py-1 "
    >
      <h1
        class="px-1 py-1"
      >
        How Well Can You Hear Audio Quality?
      </h1>
      <div
        class="flex justify-start gap-1 px-1 py-1"
      >
        <div
          class="resultTag px-1 py-1"
        >
           points
        </div>
        <div
          class="flex justify-between items-center resultTag px-1 py-1 "
        >
          <div
            class="px-1 "
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
              />
            </svg>
          </div>
          <div
            class="px-1 "
          >
            pingec
          </div>
        </div>
        <div
          class="resultTag  px-1 py-1"
        >
          over 7 years ago
        </div>
        <div
          class="resultTag px-1 py-1"
        >
           comments
        </div>
        <a
          class="resultTag  px-1 py-1"
        />
      </div>
      <div>
        Similar test: 
      </div>
    </div>
  </div>
</DocumentFragment>
`;
